docs/resources/azure_virtual_machine_disk.md

```diff
@@ -0,0 +1,117 @@
+---
+title: About the azure_virtual_machine_disk Resource
+platform: azure
+---
+
+# azure_virtual_machine_disk
+
+Use the `azure_virtual_machine_disk` InSpec audit resource to test properties and configuration of an Azure disk.
+
+## Azure REST API version, endpoint and http client parameters
+
+This resource interacts with api versions supported by the resource provider.
+The `api_version` can be defined as a resource parameter.
+If not provided, the latest version will be used.
+For more information, refer to [`azure_generic_resource`](azure_generic_resource.md).
+
+Unless defined, `azure_cloud` global endpoint, and default values for the http client will be used.
+For more information, refer to the resource pack [README](../../README.md). 
+
+## Availability
+
+### Installation
+
+This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure). 
+For an example `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).
+
+## Syntax
+
+`resource_group` and `name` or the `resource_id` must be given as a parameter.
+```ruby
+describe azure_virtual_machine_disk(resource_group: 'inspec-resource-group-9', name: 'example_disk') do
+  it { should exist }
+end
+```
+```ruby
+describe azure_virtual_machine_disk(resource_id: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks/{diskName}') do
+  it { should exist }
+end
+```
+## Parameters
+
+| Name                                  | Description                                                                       |
+|---------------------------------------|-----------------------------------------------------------------------------------|
+| resource_group                        | Azure resource group that the targeted resource resides in. `MyResourceGroup`     |
+| name                                  | Name of the disk to test. `MyDisk`                                                |
+
+Either one of the parameter sets can be provided for a valid query:
+- `resource_id`
+- `resource_group` and `name`
+
+## Properties
+
+| Property          | Description |
+|---------------------------|-------------|
+| encryption_enabled        | Indicates whether the encryption is enabled or not. Note that this will return `nil` unless the encryption status is defined on the resource explicitly. |
```

Comment:
I wonder if we could add a resource method or attribute reader, for this property and default it `false` or whatever the value should be for encryption is disabled.

Reply:
In fact, I see that you have already created a method for this. It would be good to just default the value to `false` or `disabled`

Reply:
This is created for backward compatibility The property used for making this conclusion, in the old API, is being presented through multiple properties with the latest API version. 
Considering this behavior change over time, I thought it is safer to return the value only if it is provided explicitly from the endpoint. (`properties.encryptionSettingsCollection.enabled`)

Even though returning `encryption_enabled = false` is fail-safe when it is encrypted, I thought it should be the user's responsibility to make this call by returning `encryption_enabled = nil`.

Reply:
Returning nil in this scenrio will actually break backward compatibility. The old `azurerm` resource did this
```
!!properties.encryptionSettings.enabled
```
As an arbitrary example, below is what would be returned...
```
2.7.0 :001 > foo = true
2.7.0 :002 > !!foo
 => true
2.7.0 :003 > foo = false
2.7.0 :004 > !!foo
 => false
2.7.0 :005 > foo = nil
2.7.0 :006 > !!foo
 => false
```

As you can see, in the old resource this would have actually returned `false` when `nil` was returned

Reply:
This does not take into account the [properties.encryption](https://docs.microsoft.com/en-us/rest/api/compute/disks/get#encryption) property where it shows: `Encryption property can be used to encrypt data at rest with customer managed keys or platform managed keys.`. (Available in the latest API.)

We can check this value and eventually return `false` if it doesn`t exist either. But, leaving this decision to the user still seems more accurate than making an assumption. The new resources default to the latest API version and this makes it difficult to return the same result across the different versions.